Assets/MRTK/SDK/Features/UX/Scripts/AppBar/AppBar.cs

```diff
@@ -373,8 +373,7 @@ public void OnButtonPressed(AppBarButton button)
         protected virtual void OnClickRemove()
         {
             // Set the app bar and bounding box to inactive
-            var boundsProvider = Target as IBoundsTargetProvider;
-            if (boundsProvider != null)
+            if (Target is IBoundsTargetProvider boundsProvider && !boundsProvider.IsNull())
```

Comment:
A bit confused about the !boundsProvider.IsNull() check since you didn't do this in other places.

ie in GameObjectExtension:

  -var requireComponentAttribute = attribute as RequireComponent;
  -if (requireComponentAttribute != null)

             +if (attribute is RequireComponent requireComponentAttribute)

Reply:
I did this anywhere the interface has been implemented as a `UnityEngine.Object`. In this case, `IBoundsTargetProvider` will be

![image](https://user-images.githubusercontent.com/3580640/104392644-5e427100-54f7-11eb-8095-c68bbb62b75f.png)

which are both `MonoBehaviour`s, so we have to go through the special null checking to ensure we use [Unity's custom `==`](https://blogs.unity3d.com/2014/05/16/custom-operator-should-we-keep-it/).

Reply:
For the example you gave, `RequireComponent` is a `System.Attribute`

![image](https://user-images.githubusercontent.com/3580640/104392770-a6619380-54f7-11eb-8ae1-b988b004e555.png)

so it'll work normally